types/pg/lib/type-overrides.d.ts

```diff
@@ -0,0 +1,13 @@
+import { CustomTypesConfig } from '..';
+
+type TypeParser<TOid = number, TReturn = any> = (oid: TOid) => TReturn;
+type TypeFormat = 'text' | 'binary';
+
+declare class TypeOverrides implements CustomTypesConfig {
+    constructor(types?: CustomTypesConfig);
+    setTypeParser(oid: number, format: TypeFormat, fn: TypeParser): void;
+    setTypeParser(oid: number, fn: TypeParser): void;
+    getTypeParser(oid: number, format?: TypeFormat): TypeParser;
+}
+
+export default TypeOverrides;
```

Comment:
Is there a benefit to using the declare keyword instead of exporting directly?

```suggestion
export default class TypeOverrides implements CustomTypesConfig {
    constructor(types?: CustomTypesConfig);
    setTypeParser(oid: number, format: TypeFormat, fn: TypeParser): void;
    setTypeParser(oid: number, fn: TypeParser): void;
    getTypeParser(oid: number, format?: TypeFormat): TypeParser;
}
```

Reply:
I thought using declare was necessary to avoid defining the actual class, but maybe it isn't necessary in a declaration file?

Reply:
As far as I know (which isn't much) `declare` is not necessary in a declaration file (`.d.ts`).
Perhaps it'd be good to test this to make sure?

Reply:
Having only the default export seems to enable automatic exports with these warnings for the utility types that I didn't intend to export.

```
ERROR: 3:6  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md                                                                   
ERROR: 4:6  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md                       ```

Using `export {};` does get tests to pass, if that's preferable.

Reply:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity, use a more appropriate syntax like `import foo = require("...")`.

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
@DanielRosenwasser Thanks for the pointer.